File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys
input = sys.stdin.readline

result = []
stack = []
num = 1
for i in range(int(input())):
    target = int(input())
    while num <= target:
        stack.append(num)
        result.append('+')
        num += 1

    if stack[-1] == target:
        stack.pop()
        result.append('-')    
    else :
        print('NO')
        break

if len(stack) == 0:
    for i in result:
        print(i)

File: 백준/Silver/1874. 스택 수열/스택 수열.cc
#include<iostream>
#include<stack>
#include<vector>
using namespace std;
stack<int> s;
vector<char> result;
int cnt = 1;
int n;
int main() {
    ios_base::sync_with_stdio(false); cin.tie(0);
    cin >> n;
    for (int i = 0; i < n; i++) {
        int x;
        cin >> x;
        while (cnt <= x) {
            s.push(cnt);
            cnt += 1;
            result.push_back('+');
        }
        if (s.top() == x) { 
            s.pop();
            result.push_back('-');
        }
        else { 
            cout << "NO";
            return 0;
        }
    }
    for (auto i : result) {
        cout << i << '\n';
    }
}